aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -69,16 +70,20 @@ impl<'a, S: 'a + StateView + Sync> ExecutorTask for AptosExecutorTask<'a, S> {
             .vm
             .execute_single_transaction(txn, &view.as_move_resolver(), &log_context)
```

Comment:
I mentioned it to @vusirikala today, but there are similar changes in this PR and if we can make the fix here, would be great:
current behavior below is to execute the transaction, and if materialize deltas is true, materialize immediately after. This was done for sequential execution under the assumption that aggregators never fail. If aggregators can fail, we would panic below, and we need to fix it. The simplest fix seems to provide the use_aggregator flag to execute_single_transaction - where use_aggregators should be !materialize_deltas (we can also rename materialize deltas to use_aggregators and fix the callers - just 2 places in block executor).

Reply:
I would say let's add this fix in a separate PR? Otherwise we would have to add tests for this as well, review, and it will only take longer to land this refactoring😄

Reply:
No worries, actually makes sense and Satya just sent a PR for it ^ so all good B-)